### DISCUSSION TITLE
Unselect node

### DISCUSSION THREAD

**Discussion by gilneto8** [xyflow/xyflow#1140] [Q&A] [answered]
Hey everyone,

I've searched the docs but haven't found an answer for this.

Is there any clean way to 'unselect' a node/edge? the function `onElementClick` gives me a clickable element, and I'm saving it as `selected` for editing purposes. I can also unselect using the Escape key, but I wanted to be able to click the map to unselect. Right now, I'm thinking only of using the `onClick` method in juction with the `onElementClick` and checking if both of them are fired (select the element) or if only of them is fired (unselect anything), but is a rather filthy way of inverting the logic. Is there any method I'm missing here? I'm using version `9.0.1` for now.

Thanks in advance

**Comment by moklick** [comment, marked as answer]
Hey @gilneto8 

a deselection happens when you click on the pane. You can listen to this change with the `onSelectionChange` handler. If you want to set/unset a selection manually you can use the `setSelectedElements` action `const setSelectedElements = useStoreActions(a => a.setSelectedElements)`. To unselect all you can do `setSelectedElements([])`.
